pennylane/qnn/torch.py

```diff
@@ -291,6 +295,58 @@ def _evaluate_qnode(self, x):
         }
         return self.qnode(**kwargs).type(x.dtype)
 
+    def _init_weights(
+        self,
+        weight_shapes: Dict[str, tuple],
+        init_method: Union[Callable, Dict[str, Union[Callable, Any]], None],
+    ):
+        r"""Initialize and register the weights with the given init_method. If init_method is not
+        specified, weights are randomly initialized from the uniform distribution on the interval
+        [0, 2π].
+
+        Args:
+            weight_shapes (dict[str, tuple]): a dictionary mapping from all weights used in the QNode to
+                their corresponding shapes
+            init_method (Union[Callable, Dict[str, Union[Callable, torch.Tensor]], None]): Either a
+                `torch.nn.init <https://pytorch.org/docs/stable/nn.init.html>`__ function for
+                initializing the QNode weights or a dictionary specifying the callable/value used for
+                each weight. If not specified, weights are randomly initialized using the uniform
+                distribution over :math:`[0, 2 \pi]`.
+        """
+
+        def init_weight(weight_name: str, weight_size: tuple) -> torch.Tensor:
+            """Initialize weights.
+
+            Args:
+                weight_name (str): Weight name
+                weight_size (tuple): Size of the weight
+
+            Returns:
+                torch.Tensor: Tensor containing the weights
+            """
+            if init_method is None:
+                init = functools.partial(torch.nn.init.uniform_, b=2 * math.pi)
+            elif callable(init_method):
+                init = init_method
+            elif isinstance(init_method, dict):
+                init = init_method[weight_name]
+                if isinstance(init, torch.Tensor):
+                    if tuple(init.shape) != weight_size:
+                        raise ValueError(
+                            f"The Tensor specified for weight '{weight_name}' doesn't have the"
+                            + "appropiate shape."
+                        )
+                    return init
+            # TODO: Figure out why a 1-dim Tensor is assigned to all weights with shape <= 1.
```

Comment:
Would this comment be out of date with [Tom's comment](https://github.com/PennyLaneAI/pennylane/issues/898#issuecomment-1151616249)?

Reply:
Yes, sorry. The comment has been removed.

Reply:
No problem!